src/assert_diagnostic.erl

```diff
@@ -0,0 +1,46 @@
+%% Copyright (c) Facebook, Inc. and its affiliates.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License
+
+%% See assert_diagnostic.hrl.
+
+-module(assert_diagnostic).
+
+-export([
+   check_length/2,
+   check_elements/3
+]).
+
+-define(FORMAT(T, Args), lists:flatten(io_lib:format(T, Args))).
```

Comment:
We have tried to write as little macros as possible.  Is it possible to remove this one, I understand we might need the `assertListEqual` macro, but lets remove the rest that we do not have to have and rather use functions.

Reply:
Done!